### PR TITLE
Convert Biostrings2classes.Rnw to .Rmd

### PR DESCRIPTION
**Pull request by villafup**
@Bioconductor/sweave2rmd for review

### CODE REVIEW

**Comment by jwokaty** [comment]
@villafup Please remove .Rnw file. I don't think you need to modify the DESCRIPTION because there are other PRs that add BiocStyle

**Comment by hpages** [comment]
@villafup @jwokaty Where are we standing with this PR?

**Comment by jwokaty** [comment]
We are waiting on some changes from @villafup. I'll slack him :)

File: vignettes/Biostrings2Classes.Rmd [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+
+This document briefly presents the new set of classes implemented in the
+`r Biocpkg('Biostrings')` 2 package. Like the `r Biocpkg('Biostrings')` 
+1 classes (found in `r Biocpkg('Biostrings')` v 1.4.x), they were designed to make manipulation of big
+strings (like DNA or RNA sequences) easy and fast. This is achieved by
+keeping the 3 following ideas from the `r Biocpkg('Biostrings')` 1 package: (1) use R
+external pointers to store the string data, (2) use bit patterns to
+encode the string data, (3) provide the user with a convenient class of
+objects where each instance can store a set of views *on the same* big
+string (these views being typicallythe matches returned by a search
```
Comment (jwokaty) [inline vignettes/Biostrings2Classes.Rmd:34]: space between `typicallythe`

File: vignettes/Biostrings2Classes.Rmd [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+algorithm).
+
+However, there is a flaw in the `r Biocpkg('Biostring')` class design that prevents
+the search algorithms to return correct information about the matches
+(i.e. the views) that they found. The new classes address this issue by
+replacing the `r Biocpkg('Biostrings')` class (implemented in `r Biocpkg('Biostrings')` 1) by 2 new
```
Comment (jwokaty) [inline vignettes/Biostrings2Classes.Rmd:40]: I think lines 37 and 40 (the first reference only) reference an old `BioString` class, so replace

    `r Biocpkg('Biostrings')`
    
with

    BioString

**Review by jwokaty — CHANGES_REQUESTED** [review]
@villafup Just 3 small changes.

I noticed that it doesn't currently generate the Rmd via `R CMD build` because `VignetteBuild: knitr` isn't in this version of the DESCRIPTION file; however, when it's merged, it should create the .Rmd.

It'll be ready for Herve after these changes.

**Comment by ahl27** [comment]
@hpages @jwokaty what is the status of this and #90 ? Are these still being worked on, and if not, can we merge/close them?

**Comment by hpages** [comment]
I'm surprised this vignette made it to the sweave2rmd list. It's very old and outdated and doesn't have much value anymore. It was written to document the classes that I introduced when I took over the package and transitioned it from **Biostrings** 1 to **Biostrings** 2 about 18 years ago! The only reason it's still around is that **Biostrings** doesn't have a good general vignette, so I guess having the "Biostrings 2 classes" vignette was better than nothing. However we should start thinking about addressing that @ahl27. In the mean time I think it's ok to close this unless @jwokaty or @villafup oppose. Thanks!

**Comment by ahl27** [comment]
Sounds good, I'll wait for either of them to chime in to confirm and then we can consider closing it. 

Is the other PR still something we need to merge? I can ping you on that PR so the conversation stays localized to the PR

**Pull request closed without merging**